webrtc.html

```diff
@@ -2582,6 +2636,110 @@
       </dl>
     </section>
 
+    <section>
+      <h2 id="sec.cert-mgmt">Certificate Management</h2>
+      <p>
+        The certificates that <code>RTCPeerConnection</code> instances use to
+        authenticate with peers use the <a><code>RTCCertificate</code></a>
+        interface.  These objects can be explicitly generated by applications
+        using <code>RTCPeerConnection.generateCertificate</code> and provided in
+        the <a><code>RTCConfiguration</code></a> when constructing a
+        new <code>RTCPeerConnection</code> instance.
+      </p>
+      <p>
+        The explicit certificate management functions provided here are
+        optional.  If an application does not provide
+        the <code>certificates</code> configuration option when constructing
+        an <code>RTCPeerConnection</code> a new set of certificates MUST be
+        generated by the <a>user agent</a>.  That set MUST include an ECDSA
+        certificate with a private key on the P-256 curve and a signature with a
+        SHA-256 hash.
+      </p>
+
+      <dl class="idl" title="partial interface RTCPeerConnection">
+        <dt>static Promise&lt;RTCCertificate&gt; generateCertificate
+          (AlgorithmIdentifier keygenAlgorithm)</dt>
+        <dd>
+          <p>This function takes an argument that describes how the private key associated with the certificate is generated.  The <var>keygenAlgorithm</var> argument uses the WebCrypto
+          [[!WebCryptoAPI]] <a href="https://dvcs.w3.org/hg/webcrypto-api/raw-file/tip/spec/Overview.html#dfn-AlgorithmIdentifier">AlgorithmIdentifier</a>
+          type.  The <var>keygenAlgorithm</var> value MUST be a valid argument
+            to <a href="https://dvcs.w3.org/hg/webcrypto-api/raw-file/tip/spec/Overview.html#SubtleCrypto-method-generateKey"><code>Crypto.subtle.generateKey</code></a>.</p>
```

Comment:
I think that you are incorrectly assuming that this needs to *use* the WebCrypto infrastructure in any way.  `AlgorithmIdentifier` is just a dictionary.  This is just using the already-defined syntax in WebCrypto to describe what it means.

Also, we aren't allowing applications to say anything about the signing process.  You're assuming a lot when you invent step 2 there.

I find the W3C habit of overspecificity in specs stifling.  What is wrong with describing inputs and outputs?  The question you need to ask here is this: Are you capable of taking information that identifies a key generation algorithm and either turning that into a certificate+private key or failing?  Why does it matter so much how that happens?  

The question of whether the algorithm identification is sufficiently well specified is definitely something I'm willing to engage on here.  The bit where the browser chooses the signature algorithm is here largely in response to your feedback, but I'm happy to iterate on that.  To that end, I will definitely reference the normalization algorithm, though it's really only a hint for implementations.